dataprofiler/profilers/unstructured_text_profile.py

```diff
@@ -0,0 +1,223 @@
+                                 self.__calculations,
+                                 other.__calculations)
+
+        if "vocab" in merged_profile.__calculations:
+            merged_profile.vocab = self.vocab.copy()
+            merged_profile._update_vocab(other.vocab)
+            
+        if "words" in merged_profile.__calculations:
+            merged_profile.word_count = self.word_count.copy()
+            for word in other.word_count:
+                if word in merged_profile.word_count:
+                    merged_profile.word_count[word] += other.word_count[word]
+                elif word not in self._stop_words:
+                    merged_profile.word_count[word] = other.word_count[word]
+                    
+        merged_profile.sample_size = self.sample_size + other.sample_size
+
+        return merged_profile
+
+    @property
+    def profile(self):
+        """
+        Property for profile. Returns the profile of the column.
+
+        :return:
+        """
+        profile = dict(
+            vocab=self.vocab,
+            words=list(self.word_count.keys()),
+            word_count=dict(self.word_count),
+            times=self.times,
+        )
+        return profile
+
+    @BaseColumnProfiler._timeit(name='vocab')
+    def _update_vocab(self, data, prev_dependent_properties=None,
+                      subset_properties=None):
+        """
+        Finds the unique vocabulary used in the text samples.
+
+        :param data: list or array of data from which to extract vocab
+        :type data: Union[list, numpy.array, pandas.DataFrame]
+        :param prev_dependent_properties: Contains all the previous properties
+            that the calculations depend on.
+        :type prev_dependent_properties: dict
+        :param subset_properties: Contains the results of the properties of the
+            subset before they are merged into the main data profile.
+        :type subset_properties: dict
+        :return: None
+        """
+        data_flat = list(itertools.chain(*data))
+        self.vocab = utils._combine_unique_sets(self.vocab, data_flat)
+
+    @BaseColumnProfiler._timeit(name='words')
+    def _update_words(self, data, prev_dependent_properties=None,
+                      subset_properties=None):
+        """
+        Finds the unique words and word count used in the text samples.
+
+        :param data: list or array of data from which to extract vocab
+        :type data: Union[list, numpy.array, pandas.DataFrame]
+        :param prev_dependent_properties: Contains all the previous properties
+            that the calculations depend on.
+        :type prev_dependent_properties: dict
+        :param subset_properties: Contains the results of the properties of the
+            subset before they are merged into the main data profile.
+        :type subset_properties: dict
+        :return: None
+        """
+        for word in list(" ".join(data).split(" ")):
```

Comment:
```
data = ["this is a test", "sentence"]
" ".join(data) # "This is a test sentence"
" ".join(data).split(" ") # ["this", "is", "a", "test", "sentence"]
```

I don't think you need to put it in a `list`?

Reply:
Oh, maybe. I'll try without and see what happens

Reply:
is it not costly to join everything and then split as opposed to just looping through each item in data applying split and adding to the dict?

Reply:
also, would `" ".join(data).split(" ")`  need `list()` around it as doesn't `split` create a list?

Reply:
fixed